persistence/persistence/src/main/java/one/microstream/persistence/types/StorerModeController.java

```diff
@@ -0,0 +1,254 @@
+{
+	private final static Logger logger = Logging.getLogger(StorerModeController.class);
+	
+	///////////////////////////////////////////////////////////////////////////
+	// instance fields //
+	////////////////////
+	
+	private XList<WeakReference<PersistenceStorerDeactivatable>> registry;
+	private boolean                                              enabledWrites = true;
+	private CleaningThread                                       cleaningThread;
+	private boolean                                              active;
+
+	/**
+	 * A helper thread that iterates all registered storers and removes them if
+	 * garbage collected.
+	 */
+	private static class CleaningThread extends Thread
+	{
+		
+		StorerModeController controller;
+		
+		public CleaningThread(final StorerModeController controller)
+		{
+			super("Microstream-StorerModeController");
+			this.controller = controller;
+		}
+		
+		@Override
+		public void run()
+		{
+			while(this.controller.hasRegisteredStorers() && this.controller.isActive())
+			{
+				try
+				{
+					Thread.sleep(1000);
+				}
+				catch (final InterruptedException e)
+				{
+					//no need to end thread.
+					logger.debug("Ignored an InterruptedException while sleeping!");
+				}
+				this.controller.clean();
+			}
+		}
+	}
+	
+	
+	///////////////////////////////////////////////////////////////////////////
+	// constructors //
+	/////////////////
+	
+	public StorerModeController()
+	{
+		super();
+		this.registry = BulkList.New();
+		this.active = true;
+	}
+	
+	
+	///////////////////////////////////////////////////////////////////////////
+	// methods //
+	////////////
+	
+	/**
+	 * Returns the state of this StorerModeController.
+	 * 
+	 * @return true if not shut down.
+	 */
+	public synchronized boolean isActive()
+	{
+		return this.active;
+	}
+
+	/**
+	 * Shutdown this StorerModeController.
+	 */
+	public synchronized void shutdown()
+	{
+		this.active = false;
+	}
+	
+	@Override
+	public synchronized PersistenceStorerDeactivatable register(final PersistenceStorerDeactivatable deactivatableStorer)
+	{
+		this.registry.add(new WeakReference<>(deactivatableStorer));
+		deactivatableStorer.setWriteEnabled(this.enabledWrites);
+		this.startCleaningTask();
+		return deactivatableStorer;
+	}
+	
+	@Override
+	public synchronized boolean hasRegisteredStorers()
+	{
+		return this.registry.size() > 0;
+	}
+		
+	@Override
+	public synchronized void clean()
+	{
+		final XList<WeakReference<PersistenceStorerDeactivatable>> newRegistry = BulkList.New();
+		
+		this.registry.forEach( e ->
+		{
+			final Object referent = e.get();
+			if(referent!=null)
+			{
+				newRegistry.add(e);
+			}
+		});
+		
+		this.registry = newRegistry;
+		logger.trace("Active storers after cleanup: {}", this.registry.size());
+	}
+	
+	/**
+	 * Switch all registered storers to the writing mode.
+	 * 
+	 * @return true if writing is enabled, otherwise false.
+	 */
+	public synchronized boolean enableWriting()
```

Comment:
To be consistent, getter/setter style methods would be best.

Reply:
You're right. Unified API.